src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -174,6 +174,7 @@ public class FailoverConnectionPlugin implements IConnectionPlugin {
 
   private long invokeStartTimeMs;
   private long failoverStartTimeMs;
+  boolean initialConnectionToReader;
```

Comment:
Nit - The naming of this boolean variable could be more concise and named in a form of a statement. For example `isInitialConnectionToReader`. Otherwise, without looking at the type, it could be confused as a connection object.